accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,416 @@
+Questions that need to be answered by the cluster provider:
+- Upon successful bootstrapping, can the worker nodes be managed by Uyuni/SUSE Manager?
+- Do the nodes have a specific agent running on them and/or can we install a `salt-minion` or use `salt-ssh` on those nodes?
+- Can the nodes be managed by Uyuni/SUSE Manager? Can the packages be updated or do they require specific handling by the Management Node?
+
+### Manage a cluster using Uyuni/SUSE Manager
+
+Uyuni/SUSE Manager offers to show the status of the running cluster by either accessing the cluster directly or via the Management Node.
+In this regard, the cluster provider should provide a way to expose the cluster status (e.g. running nodes, the health of each node, monitoring metrics...) to Uyuni/SUSE Manager.
+
+#### Add and removal of nodes to the cluster
+
+Uyuni/SUSE Manager also offers the possibility of adding and removing nodes to the cluster:
+
+- To add a node from scratch, an AutoYaST profile can be reused
+- To remove a node, Uyuni/SUSE Manager must remove the node from the cluster using the Management Node and issuing the proper procedure
+
+Uyuni/SUSE Manager should be instructed to perform additional actions depending on the role of added/removed node (e.g. control plane, worker, load balancer, ...).
+
+#### Cluster upgrade
+
+If every node of the cluster is not directly registered to Uyuni/SUSE Manager, Uyuni/SUSE Manager should offer the possibility to use the Management Node to trigger an update of every node in the cluster, following the standard procedure of upgrading predisposed by the cluster provider.
+
+Also, Uyuni/SUSE Manager must coordinate the Management Node to issue a correct upgrade procedure of the product (e.g. from CaaS Platform version 3 to 4): this process is initiated by Uyuni/SUSE Manager by using the cluster provider framework implementation (e.g. `skuba`) to handle the upgrade operation.
+
+#### Application deployment
+
+Uyuni/SUSE Manager can also offer the possibility of deploying an application on top of the cluster deployed (e.g. Kubernetes deployment descriptor to deploy a Kubernetes operator). In this case, the cluster provider must provide the Management Node of the necessary tooling to deploy an application using an API. Uyuni/SUSE Manager makes use of the Management Node to deploy a provided application.
+
+#### Specific usage of Uyuni/SUSE Manager features
+
+One cluster deployment can also leverage Uyuni/SUSE Manager features via API.
+For example, a cluster deployment may invoke `CVEAuditHandler` and test if a container image built by SUSE Manager is vulnerable against a particular CVE and decide whether to allow the deployment of the image or not.
+
+## CaaS Platform-specific design
+
+It is intended that Uyuni/SUSE Manager will not deal and will not know with the application deployed on top of CaaS Platform.
+
+### Deploy a CaaS platform cluster using Uyuni/SUSE Manager
+
+  #### Cluster Deployment
```

Comment:
I think this and other subtitles are not going to be rendered properly because they are indented.

Reply:
Fixed